stable/redis/templates/deployment.yaml

```diff
@@ -23,6 +23,8 @@ spec:
             secretKeyRef:
               name: {{ template "fullname" . }}
               key: redis-password
+        - name: ALLOW_EMPTY_PASSWORD
```

Comment:
@tompizmor I believe this isn't needed by default since we do pass a password.

There is a separate issue about having a value to run Redis without a password (#1061). Perhaps we should do this in this PR?